ftplugin/NeogitCommitMessage.lua

```diff
@@ -0,0 +1,16 @@
+if vim.b.did_ftplugin then
+  return
+end
+
+vim.cmd.source("$VIMRUNTIME/ftplugin/gitcommit.vim")
+vim.cmd([[exec 'norm! gg']])
+
+if vim.fn.prevnonblank(".") ~= vim.fn.line(".") then
+  vim.cmd([[startinsert]])
```

Comment:
Does this respect the config option `disable_insert_on_commit`?

Reply:
No.. Probably should. Or just be sure to call `stopinsert` if it's set to true in the buffer constructor.

Reply:
startinsert is already called in [`commit_](https://github.com/TimUntersberger/neogit/blob/c8af6d8429ba5b4bd5686084bf70d0d1483d0a18/lua/neogit/buffers/commit_editor/init.lua#L77)

So I do not think it is needed here

Reply:
I do not think `stopinsert` is the right either as we will have called insert enter autocommands, for a short while. Those could then have `vim.schedule`d something which will break in the next tick as we are no longer in insert mode.

I think it will cause more harm than good when considering that insert modes changes tend to lead to many interactions with other plugins. I know of quite a few other plugins who have had odd interactions due to there functions.

Reply:
Hmm. Alright, great points. Maybe a better course would be to add this as a config option, ie, `smart_insert` and move it into the buffer constructor.

Reply:
It is already there in the constructor :)

On another note: I'd really prefer if insert_on_commit is left as optional since I prefer to not enter insert mode automatically

Reply:
Totally fair - just haven't had time to fix up this PR yet. I'll try to get that all working tonight :)